prometheus/metric.go

```diff
@@ -187,6 +187,12 @@ func (m *withExemplarsMetric) Write(pb *dto.Metric) error {
 		pb.Counter.Exemplar = m.exemplars[len(m.exemplars)-1]
 	case pb.Histogram != nil:
 		for _, e := range m.exemplars {
+			if pb.Histogram.Schema != nil {
+				if *pb.Histogram.Schema > math.MinInt32 && e.GetTimestamp() != nil {
+					pb.Histogram.Exemplars = append(pb.Histogram.Exemplars, e)
+				}
+				continue
```

Comment:
If a classic and a native histogram is exposed at the same time (for the scraper to pick), we should have the exemplars on both. Maybe the following works for that:

```suggestion
				if len(pb.Histogram.Bucket) == 0 {
					// Don't proceed to classic buckets if there are none.
					continue
				}
```

We do have to check if there are already buckets defined, because otherwise there will be an Inf bucket created just for the histograms.